### PR TITLE
move selectorsToReason into selectors

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3993

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by meain** [comment]
This will need a rebase before I could do a review. The diff contains a lot of things that are not part of the change.

**Review by meain — APPROVED** [review]
Code looks good, but I'm worried whether we are over-complicating the implementation. I might be missing something, but I'm not sure what advantage having a `reasoner` interface would improve, at least as of now.

Approving to unblock.

**Comment by ryanfkeepers** [comment]
>  I'm worried whether we are over-complicating the implementation. I might be missing something, but I'm not sure what advantage having a reasoner interface would improve, at least as of now.

The reasoner interface has existed for a while now.  We're just looking at various ways to comply with it.  I think the pieces will fit together more obviously as you look at the path interface changes.

File: src/internal/operations/backup.go [hunk 1]
```diff
@@ -424,13 +430,14 @@ func (op *BackupOperation) do(
 	return deets, nil
 }
 
-func makeFallbackReasons(tenant string, sel selectors.Selector) []identity.Reasoner {
+func makeFallbackReasons(tenant string, sel selectors.Selector) ([]identity.Reasoner, error) {
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/operations/backup.go:433]: Preferably not, since we plan to deprecate this once we can drop the v0.7.0 migration support.

**Pull request merged**